### PR TITLE
Rails 5 upgrade

### PR DESCRIPTION
**Pull request by dwhenry**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -24,9 +24,9 @@ gem 'notifications-ruby-client', github: 'alphagov/notifications-ruby-client'
 gem 'output-templates', '~> 4.4.4', github: 'guidance-guarantee-programme/output-templates'
 gem 'pg'
 gem 'plek'
-gem 'princely'
+gem 'princely', github: 'guidance-guarantee-programme/princely', branch: 'remove-alias-method-chain'
 gem 'puma'
-gem 'rails', '4.2.6'
+gem 'rails', '5.0.0.1'
```
Comment (benlovell) [inline Gemfile]: Can we also bump this to 5.0.1 now?

File: config/environments/development.rb [hunk 2]
```diff
@@ -10,13 +9,28 @@
...
+  # Show full error reports.
   config.consider_all_requests_local = true
-  config.action_controller.perform_caching = false
+
+  # Enable/disable caching. By default caching is disabled.
+  if Rails.root.join('tmp/caching-dev.txt').exist?
+    config.action_controller.perform_caching = true
+
+    config.cache_store = :memory_store
+    config.public_file_server.headers = {
```
Comment (benlovell) [inline config/environments/development.rb]: We ought to add this to the production environment too.

File: db/migrate/20161207095801_remove_permissions_default_value.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class RemovePermissionsDefaultValue < ActiveRecord::Migration[5.0]
+  def change
+    change_column :users, :permissions, :string, default: nil
```
Comment (benlovell) [inline db/migrate/20161207095801_remove_permissions_default_value.rb]: Why did we need to do this?

Comment (dwhenry) [inline db/migrate/20161207095801_remove_permissions_default_value.rb]: It's to do with the YAML deserialising. 
Without the field defaults to an empty string which does NOT deserialise to an Array as expected.

File: config/environments/development.rb (hunk 2 above)
Comment (dwhenry) [inline config/environments/development.rb]: `public_file_server` is not being used in prod as `RAILS_SERVE_STATIC_FILES` is not set.

File: Gemfile (hunk 1 above)
Comment (dwhenry) [inline Gemfile]: doing..

File: config/environments/development.rb (hunk 2 above)
Comment (benlovell) [inline config/environments/development.rb]: How are we serving the assets?

File: db/migrate/20161207095801_remove_permissions_default_value.rb (hunk 3 above)
Comment (benlovell) [inline db/migrate/20161207095801_remove_permissions_default_value.rb]: Did this manifest itself in the specs?

Comment (dwhenry) [inline db/migrate/20161207095801_remove_permissions_default_value.rb]: yes

Comment (benlovell) [inline db/migrate/20161207095801_remove_permissions_default_value.rb]: Cool.

**Comment by dwhenry** [comment]
@benlovell all done including use of ApplicationRecord.

**Comment by dwhenry** [comment]
@benlovell all changes made as requested.

**Pull request merged**